CONTRIBUTING.md

```diff
@@ -135,6 +135,15 @@ The conventions for module exports are:
 
 * We use [rebase merging](https://git-scm.com/book/en/v2/Git-Branching-Rebasing) (as opposed to [basic merging](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging#Basic-Merging)) to merge branches
 
+Here is a recommended way to get setup:
+1. Fork this project
+2. Clone your new fork, `git clone git@github.com:GithubUser/mapbox-gl-js.git`
+3. `cd mapbox-gl-js`
+4. Add the Mapbox repository as an upstream repository: `git add remote upstream git@github.com:mapbox/mapbox-gl-js.git`
+5. Create a new branch `git checkout -b your-branch` for your contribution
+6. Write code, open a PR from your branch when you're ready
+7. When you need to rebase your fork's PR branch onto master: `git fetch upstream`, `git rebase upstream/master` and force push to Github `git push --force origin your-branch`
```

Comment:
Overall I think this is helpful for contributors, when I first started this was a paint point for me too.

Point 7, maybe better to use "If you need to. rebase ... to resolve conflicts:" to help make it clearer that most of the time you don't need to rebase, it's only if other conflicting commits have happened since you branched. I wouldn't want people to think they have to constantly rebase and force push even when there are no conflicts.